### PR TITLE
completing data pipeline for ad hoc recommendations (for testing)

### PR DESCRIPTION
**Pull request by stkem**
@lawzlo @yingjieMiao

### CODE REVIEW

File: kifi-backend/modules/curator/app/com/keepit/curator/commanders/RecommendationGenerationCommander.scala [hunk 1]
```diff
@@ -17,17 +17,17 @@ class RecommendationGenerationCommander @Inject() (
     scoringHelper: UriScoringHelper) {
 
   def getAdHocRecommendations(userId: Id[User], howManyMax: Int): Future[Seq[Recommendation]] = {
-    val seedsFuture = seedCommander.getRecentItems(userId, howManyMax)
+    val seedsFuture = seedCommander.getTopItems(userId, Math.max(howManyMax, 250))
```
Comment (yingjieMiao) [inline kifi-backend/modules/curator/app/com/keepit/curator/commanders/RecommendationGenerationCommander.scala:20]: `250 max howManyMax` works as well

File: kifi-backend/modules/curator/app/com/keepit/curator/commanders/SeedIngestionCommander.scala [hunk 2]
```diff
@@ -94,4 +94,22 @@ class SeedIngestionCommander @Inject() (
...
   }
 
+  //this is a methods for (manual, not unit) testing of data flow and scoring, not meant for user facing content or scale
+  def getTopItems(userId: Id[User], howManyMax: Int): Future[Seq[SeedItem]] = {
+    //gets higest scoring ruis for the user. If that's not enough get recent items as well
+    db.readOnlyReplicaAsync { implicit session =>
+      var items = rawSeedsRepo.getByTopPriorScore(userId, howManyMax)
+      if (items.length < howManyMax) items = (items ++ rawSeedsRepo.getRecent(userId, howManyMax)).toSet.toSeq
+      items.map { rawItem =>
+        val keepers = if (rawItem.timesKept > 100) {
```
Comment (yingjieMiao) [inline kifi-backend/modules/curator/app/com/keepit/curator/commanders/SeedIngestionCommander.scala]: maybe put this const in `SeedItem`'s companion object?

**Comment by yingjieMiao** [comment]
a few minor comments. but good to merge.

File: kifi-backend/modules/curator/app/com/keepit/curator/commanders/RecommendationGenerationCommander.scala (hunk 1 above)
Comment (stkem) [inline kifi-backend/modules/curator/app/com/keepit/curator/commanders/RecommendationGenerationCommander.scala:20]: ah, I didn't know you could do that. On the other hand, having `max` as an infix operator is kind of un-mathematical. I think I'll leave this one.

File: kifi-backend/modules/curator/app/com/keepit/curator/commanders/SeedIngestionCommander.scala (hunk 2 above)
Comment (stkem) [inline kifi-backend/modules/curator/app/com/keepit/curator/commanders/SeedIngestionCommander.scala]: good point. will move that.

**Pull request merged**